brian2/parsing/rendering.py

```diff
@@ -287,15 +293,22 @@ class CPPNodeRenderer(NodeRenderer):
           # Bool ops
           'And': '&&',
           'Or': '||',
+          # C does not have a floor division operator (but see render_BinOp)
+          'FloorDiv': '/',
           })
     
     def render_BinOp(self, node):
-        if node.op.__class__.__name__=='Pow':
+        if node.op.__class__.__name__ == 'Pow':
             return '_brian_pow(%s, %s)' % (self.render_node(node.left),
                                     self.render_node(node.right))
-        elif node.op.__class__.__name__=='Mod':
+        elif node.op.__class__.__name__ == 'Mod':
             return '_brian_mod(%s, %s)' % (self.render_node(node.left),
                                            self.render_node(node.right))
+        elif node.op.__class__.__name__ == 'Div':
+            # C uses integer division, this is a quick and dirty way to assure
+            # it uses floating point division for integers
+            return '1.0f*%s/%s' % (self.render_element_parentheses(node.left),
+                                   self.render_element_parentheses(node.right))
```

Comment:
I don't think we ever fixed https://github.com/brian-team/brian2/issues/661 did we? If so, perhaps this PR is the right time to do that? The code above for Div is correct, but we'd (I guess) need to do something a bit more for FloorDiv since ``-1//2==-1`` in Python but ``-1/2==0`` in C.

Reply:
This PR actually has the side effect of fixing #661 in a few situations. But you are right, we still have to do something for C.